server/controllers/industry.js

```diff
@@ -1,27 +1,143 @@
+const jwt = require('jsonwebtoken');
 const User = require('../models/user');
-const JWT = require('jsonwebtoken');
+const IndustryData = require('../models/industryData');
+const { jobSchema } = require('../helpers/inputValidation/requestValidation');
 const generateRes = require('../helpers/generateJSON');
 
-const getName = async (req, res) => {
+// Get all the jobs associated with an industry user.
+const getJobs = async (req, res) => {
     try {
         const accessToken = req.header('Authorization').split(' ')[1];
-        const decodeAccessToken = JWT.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
 
         //check if user exists
         const user = await User.findOne({ email: decodeAccessToken.email });
 
         //check if user type is industry
-        if (user.userType.Type == process.env.INDUSTRY) {
-            return res.status(200).json(generateRes(true, 200, "INDUSTRY_USER_NAME", {
-                name: user.name,
-            }));
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, 'BAD_REQUEST', {}));
+        }
+
+        const industryData = await IndustryData.findById(user.userType.industryData);
+        if (!industryData) {
+            return res.status(404).json(generateRes(false, 404, "INDUSTRY_DATA_NOT_FOUND", {}));
         }
-        return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", {}));
+
+        const jobs = {
+            active: industryData.jobs.active,
+            draft: industryData.jobs.draft,
+            archived: industryData.jobs.archived,
+        };
+
+        return res.status(200).json(generateRes(true, 200, "JOBS_FOUND", { jobs: jobs }));
     } catch (error) {
         return res.status(400).json(generateRes(false, 400, "BAD_REQUEST", { error }));
     }
+
 }
 
+// Create a job with the data given in the request payload
+const createJob = async (req, res) => {
+    try {
+        const accessToken = req.header('Authorization').split(' ')[1];
+        const decodeAccessToken = jwt.verify(accessToken, process.env.SECRET_ACCESS_TOKEN);
+
+        // check if user exists
+        const user = await User.findOne({ email: decodeAccessToken.email });
+
+        // check if user type is industry
+        if (user.userType.Type != process.env.INDUSTRY) {
+            return res.status(400).json(generateRes(false, 400, 'BAD_REQUEST', {}));
+        }
+
+        const jobType = req.body.jobType;
+
+        // validate schema
+        const { error } = jobSchema.validate(req.body.jobDetails);
+        if (error) {
+            return res.status(400).json(generateRes(false, 400, "INPUT_ERROR", error));
+        }
+        if (!(jobType === "active" || jobType === "draft")) {
```

Comment:
Not anything you need to change, but we should add environment variables for the jobtypes to avoid this having lowercase while the faculty projects have upper case